Sources/Scout/Definitions/Path+Flattening.swift

```diff
@@ -0,0 +1,183 @@
+//
+// Scout
+// Copyright (c) Alexis Bridoux 2020
+// MIT license, see LICENSE file for details
+
+import Foundation
+
+// MARK: - Storage
+
+private struct IndexedSlice {
```

Comment:
A lot of structures within a single file, it would be better to put them in their own files.

Reply:
If they were to be put in their own file, I would have to make them internal at least. However, they are only use for the flattening functions to make the code clearer. Thus, maybe it's better if they remain in the same file since they are not used elsewhere.